### PR TITLE
HIVE-24810: Use JDK 8 String Switch in TruncDateFromTimestamp

### PR DESCRIPTION
**Pull request by belugabehr**

### CODE REVIEW

**Comment by belugabehr** [comment]
@miklosgergely Minor improvement if you have any PR review cycles.

**Comment by belugabehr** [comment]
@miklosgergely  :)

**Comment by belugabehr** [comment]
@pgaref If you are available for a quick review :)

File: ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/TruncDateFromTimestamp.java [hunk 1]
```diff
@@ -147,18 +148,29 @@ protected void truncDate(ColumnVector inV, BytesColumnVector outV, int i) {
   }
 
   protected void processDate(BytesColumnVector outV, int i, Date date) {
-    if ("MONTH".equals(fmt) || "MON".equals(fmt) || "MM".equals(fmt)) {
+    switch (fmt) {
+    case "YEAR":
+    case "YYYY":
+    case "YY":
+      date.setMonth(1);
```
Comment (pgaref) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/TruncDateFromTimestamp.java:155]: date.setMonth(1) might be missing here?

Comment (belugabehr) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/TruncDateFromTimestamp.java:155]: @pgaref Thanks so much for taking a look.  I'm not sure however what your comment means.

The year stuff get `Month=1` and `dayOfMonth=1`.  The Month stuff gets only `dayOfMonth=1`.

This switch statement "falls through" to cover both outcomes.

Comment (pgaref) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/TruncDateFromTimestamp.java:155]: Hey @belugabehr  -- it was actually the fall through that confused me, I see what you are doing now
In the original call it was part of an else if statement taking care of both.

**Review by pgaref — APPROVED** [review]

**Pull request merged**